drivers/timer/stm32_lptim_timer.c

```diff
@@ -0,0 +1,285 @@
+/*
+ * Copyright (c) 2018 Foundries.io Ltd
+ * Copyright (c) 2019 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <soc.h>
+#include <drivers/clock_control.h>
+#include <drivers/clock_control/stm32_clock_control.h>
+#include <drivers/timer/system_timer.h>
+#include <sys_clock.h>
+
+#include <spinlock.h>
+
+/*
+ * Assumptions and limitations:
+ *
+ * - system clock based on an LPTIM1 instance, clocked by LSI or LSE
+ * - prescaler is set to 1 (LL_LPTIM_PRESCALER_DIV1 in the related register)
+ * - using LPTIM1 AutoReload capability to trig the IRQ (timeout irq)
+ * - when timeout irq occurs the counter is already reset
+ * - the maximum timeout duration is reached with the LPTIM_TIMEBASE value
+ * - with prescaler of 1, the max timeout (LPTIM_TIMEBASE) is 2seconds
+ */
+
+#define LPTIM_CLOCK CONFIG_STM32_LPTIM_CLOCK
+#define LPTIM_TIMEBASE CONFIG_STM32_LPTIM_TIMEBASE
+
+/* nb of LPTIM counter unit per ms */
+#define COUNT_PER_MS (LPTIM_CLOCK / 1000)
+
+/* nb of LPTIM counter unit per kernel tick  */
+#define COUNT_PER_TICK (LPTIM_CLOCK / CONFIG_SYS_CLOCK_TICKS_PER_SEC)
+
+static u32_t last_cnt;
+static u32_t accumulated_lptim_ms;
+
+static struct k_spinlock lock;
+
+static void lptim_irq_handler(struct device *unused)
+{
+	ARG_UNUSED(unused);
+
+	if ((LL_LPTIM_IsActiveFlag_ARRM(LPTIM1) != 0)
+		&& (LL_LPTIM_IsEnabledIT_ARRM)(LPTIM1) != 0) {
+
+		k_spinlock_key_t key = k_spin_lock(&lock);
+
+		/* LPTIM1 CNT could have already been reset */
```

Comment:
Comment is not so clear, what are we storing in cnt_value ?
Could CNT be reset already, or has it for sure been reset already ?